### ISSUE TITLE
Spectral error for invalid paths is hard to understand

### ISSUE THREAD

**Issue by dijitali** [stoplightio/spectral#1303]
**Describe the bug**
The Spectral error if an invalid path is used is a bit cryptic to understand.

**To Reproduce**

1. Given this OpenAPI/AsyncAPI document: https://petstore.swagger.io/v2/swagger.yaml
2. Set an invalid path value on line 35 by removing the preceeding`/`:
    ```yaml
    paths:
      pet/{petId}/uploadImage:
    ```
2. Run this CLI command `spectral lint swagger.yaml`
3. See error:
``` 
OpenAPI 2.0 (Swagger) detected

/tmp/swagger.yaml
  34:7     error  oas2-schema             should NOT have additional properties.
...
```

**Expected behavior**
The Swagger Editor gives a clearer indication of the problem, it would be useful if Spectral provided something similar:

> Structural error at paths
> should only have path names that start with `/`

**Screenshots**
Swagger editor:
![image](https://user-images.githubusercontent.com/4665045/90157542-c1cc4500-dd85-11ea-81db-da4a49b88b15.png)

**Environment (remove any that are not applicable):**
 - Library version: 5.4.0
 - OS: macOS

**Comment by philsturgeon** [comment]
@P0lip can you see if your error swallowing changes are going to fix this one too? It seems like it might.

**Comment by philsturgeon** [comment]
Loads of improvements to paths in 5.x and 6.0 so let’s close this.
